src/main/java/org/testng/internal/ExitCodeListener.java

```diff
@@ -0,0 +1,69 @@
+package org.testng.internal;
+
+import org.testng.IReporter;
+import org.testng.ISuite;
+import org.testng.ISuiteResult;
+import org.testng.ITestContext;
+import org.testng.ITestListener;
+import org.testng.ITestResult;
+import org.testng.xml.XmlSuite;
+
+import java.util.List;
+
+public class ExitCodeListener implements ITestListener, IReporter {
+    private boolean hasTests = false;
+    private ExitCode status = new ExitCode();
```

Comment:
can be final

Reply:
Fixed. Don't know how that one slipped :)